### PR TITLE
feat(subscriptions) Add flag used for legacy subscriptions

### PR DESCRIPTION
**Pull request by evanh**
Subscriptions should not be created with time conditions, so add a flag to the
legacy parser that skips time conditions for when Sentry uses the SDK to convert
subscriptions.

### CODE REVIEW

File: snuba_sdk/legacy.py [hunk 1]
```diff
@@ -242,8 +244,9 @@ def json_to_snql(body: Mapping[str, Any], entity: str) -> Query:
                     Condition(Column(time_column), op, parse_datetime(date_val))
                 )
 
-    if body.get("project"):
-        proj_cond = parse_extension_condition("project_id", body["project"], True)
+    project_id = body.get("project") or body.get("project_id")
+    if project_id:
+        proj_cond = parse_extension_condition("project_id", project_id, True)
```
Comment (fpacifici) [inline snuba_sdk/legacy.py:249]: Why this change ?

File: snuba_sdk/legacy.py [hunk 2]
```diff
@@ -233,7 +235,7 @@ def json_to_snql(body: Mapping[str, Any], entity: str) -> Query:
 
     assert isinstance(query.match, Entity)
     time_column = get_required_time_column(query.match.name)
-    if time_column:
+    if time_column and not skip_time:
```
Comment (fpacifici) [inline snuba_sdk/legacy.py:238]: I think we should reject a query that tries to set time range conditions for subscriptions. Not silently drop those conditions.

File: tests/test_legacy_subscriptions.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            (
+                "WHERE project_id IN tuple(2) "
+                "AND type = 'error' "
+                "AND ifNull(tags[level], '') = 'error' "
+                "AND environment = 'development'"
+            ),
+        ),
+        "events",
+        id="basic_subscription_with_extra_conditions",
+    ),
```
Comment (fpacifici) [inline tests/test_legacy_subscriptions.py:88]: Am I missing something or there is no test that is actually asserting that time conditions in the legacy query are skipped ?

File: snuba_sdk/legacy.py (hunk 1 above)
Comment (evanh) [reply snuba_sdk/legacy.py:249]: Subscriptions use `project_id` as a key instead of `project`.

File: snuba_sdk/legacy.py (hunk 2 above)
Comment (evanh) [reply snuba_sdk/legacy.py:238]: This isn't dropping anything. It is preventing the SDK from automatically adding them.

File: tests/test_legacy_subscriptions.py (hunk 3 above)
Comment (evanh) [reply tests/test_legacy_subscriptions.py:88]: The fact that the time conditions are not in the generated SnQL query is the test.

File: snuba_sdk/legacy.py (hunk 2 above)
Comment (fpacifici) [reply snuba_sdk/legacy.py:238]: Can't the client decide not to add `from_date` or `to_date` to the query body?
The loop below only adds the conditions if those fields are in the query body in the first place.
If no `from_date` and `to_date` are present it would not add anything, so `if skip_time`, we would ignore those fields.

Shouldn't we reject the query if the user provided `from_date` and `to_date` in the legacy query ?

Comment (evanh) [reply snuba_sdk/legacy.py:238]: Oh I see what you're saying now. You're right this change isn't needed. If someone tries to add a time condition to a subscription, then the validation in Snuba will catch it and error. So the SDK can work as normal.

**Comment by evanh** [comment]
I'm going to close this PR because I don't think it's necessary. The `skip_time` field isn't very helpful since the conditions are already skipped if `from_date` and `to_date` are missing from the body, so the SDK will work as expected in that case. I will change Sentry to pass the project ID in the `project` field to fix that issue.
